pkg/interface/chat/src/js/reducers/update.js

```diff
@@ -29,6 +31,23 @@ export class UpdateReducer {
     }
   }
 
+  reducePending(message, state) {
+    if (message) {
+      for (let pendingMessage of state.pendingMessages) {
+
+        // Does the incoming message match a pending one?
+        if (message.envelope.gam.uid === pendingMessage.uid) {
+
+          // Mutating the pendingMessages array.
+          let pendingMessageArray = state.pendingMessages;
+          let matchedMessage = _.invert(pendingMessageArray)[pendingMessage];
```

Comment:
If you're having to do inverts, it's a good sign you might want to be using an object of arrays instead of an array. This is expensive.